lib/hanami/model/associations/dsl.rb

```diff
@@ -0,0 +1,29 @@
+module Hanami
+  module Model
+    module Associations
+      # Auto-infer relations linked to repository's associations
+      #
+      # @since x.x.x
+      # @api private
+      class Dsl
```

Comment:
nitpicky but can this be `DSL`? It's an acronym so we should capitalize if we can